internal/session.go

```diff
@@ -288,12 +288,16 @@ func createSession(ctx Context, creationTaskqueue string, options *SessionOption
 	}
 
 	taskqueueChan := GetSignalChannel(ctx, sessionID) // use sessionID as channel name
-	// Retry is only needed when creating new session and the error returned is NewApplicationError(errTooManySessionsMsg)
+	// Retry is only needed when creating new session and the error returned is
+	// NewApplicationError(errTooManySessionsMsg). Therefore we make sure to
```

Comment:
Please mark this error as retryable on L530.

Reply:
Thanks! I may need a test here

Reply:
I have fixed this and added a test to confirm. I inadvertently learned that we can't call `workflow.CompleteSession` in a separate `workflow.Go` coroutine.